transactions/epoch/admin/recover_epoch.cdc

```diff
@@ -0,0 +1,48 @@
+import FlowEpoch from "FlowEpoch"
+import FlowIDTableStaking from "FlowIDTableStaking"
+import FlowClusterQC from "FlowClusterQC"
+
+// The recoverEpoch transaction creates and starts a new epoch in the FlowEpoch smart contract
+// to which will force the network exit EFM. The recoverEpoch service event will be emitted 
+// and processed by all protocol participants and each participant will update their protocol 
+// state with the new Epoch data.
+// This transaction should only be used with the output of the bootstrap utility:
+//   util epoch efm-recover-tx-args
+transaction(startView: UInt64,
```

Comment:
We should include the `epochCounter` of the `RecoveryEpoch` here and use it for safety checks in the `recoverNewEpoch` implementation. 

In `util epoch efm-recover-tx-args`, we know exactly what the `RecoveryEpoch` epoch counter must be to be accepted by the Protocol State (current+1). If we don't pass that counter value here, then we're relying on the Protocol/SmartContract counters being synchronized in a particular way for correctness. In EFM, that is not guaranteed.

Currently we set the `RecoveryEpoch` counter as follows:
- if `initNewEpoch` then `RecoveryEpoch.counter` = currentEpochCounter+1
- if `!initNewEpoch` then `RecoveryEpoch.counter` = currentEpochCounter

I would change this to:
- pass in `recoveryEpochCounter` as an argument
- change `initNewEpoch` to be `unsafeAllowOverwrite`

In `recoverNewEpoch`, check that `recoveryEpochCounter == currentEpochCounter+1`.
- If that passes, then continue
- If that fails, then only continue if `unsafeAllowOverwrite` is set to true.

Reply:
https://github.com/onflow/flow-core-contracts/commit/1e41ecf0ca8519eec08751639a3ad3928072c8a0